src/Symfony/Component/Amqp/RetryStrategy/ExponentialRetryStrategy.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+namespace Symfony\Component\Amqp\RetryStrategy;
+
+/**
+ * The retry mechanism is based on a truncated exponential backoff algorithm.
+ */
+class ExponentialRetryStrategy implements RetryStrategyInterface
+{
+    private $max;
+    private $offset;
+
+    /**
+     * @param int $max The maximum number of time to retry (0 means indefinitely)
+     * @param int $max The offset for the first power of 2
+     */
+    public function __construct($max = 0, $offset = 0)
+    {
+        $this->max = $max;
+        $this->offset = $offset;
+    }
+
+    public function isRetryable(\AMQPEnvelope $msg)
```

Comment:
What's about passing an Exception or a Context about the need to retry a message to let the strategy choose a better response.